src/main/scala/is/hail/distributedmatrix/BlockMatrix.scala

```diff
@@ -604,15 +618,31 @@ class BlockMatrix(val blocks: RDD[((Int, Int), BDM[Double])],
       .map { case (i, a) => IndexedRow(i, new DenseVector(a)) },
       nRows, nCols)
   }
-  
+
   def filterRows(keep: Array[Long]): BlockMatrix = this.transpose().filterCols(keep).transpose()
 
   def filterCols(keep: Array[Long]): BlockMatrix =
     new BlockMatrix(new BlockMatrixFilterColsRDD(this, keep), blockSize, nRows, keep.length)
-  
+
   def filter(keepRows: Array[Long], keepCols: Array[Long]): BlockMatrix =
     new BlockMatrix(new BlockMatrixFilterRDD(this, keepRows, keepCols),
       blockSize, keepRows.length, keepCols.length)
+
+
+  def melt(hc: HailContext): Table = {
+    // melt the BlockMatrix into a RDD[Row] where Row has the form (Long, Long, Double)
+    val melted_rdd = this.blocks.flatMap { case (blockId, dense_matrix) =>
```

Comment:
You can use patten matching on the inner tuples as well to avoid ._1 notation. As written, you may overflow before .toLong is applied. You can pull the multiplication outside the iterator. E.g.:
```
blocks.flatMap { case ((blockRow, blockCol), block) =>
  val rowOffset = blockRow * blockSize.toLong
  val colOffset = blockCol * blockSize.toLong
  block.activeIterator
    .map { case ((i, j), entry) =>
      Row(rowOffset + i, colOffset + j, entry)
    }
}
```

We enforce that the blocks in a block matrix are represented compactly: all elements of the data array are entries of the matrix, either column-major if isTranspose=true and rowMajor if isTranspose = false. `activeIterator` will be slower than what you'd get by building your iterator more directly to pass over the data array. I don't think it will matter much overall though since the savings is likely small compared to the other work you're doing per element here. Just noting it in case it's a bottleneck later.

More critically, you should construct an RDD of RegionValue (each a struct with two longs and a double) and call `new Table(hc, rdd, rvRowType` (I think @jigold forgot the `new`). Otherwise you're building these Row objects only to then convert them. Let us know if you have questions on how to do this!